diff -urN tvheadend/Makefile.ffmpeg tvheadend-ffmpeg6/Makefile.ffmpeg
--- tvheadend/Makefile.ffmpeg	2023-04-30 11:58:04.668159236 -0700
+++ tvheadend-ffmpeg6/Makefile.ffmpeg	2023-04-30 11:59:41.000000000 -0700
@@ -108,10 +108,10 @@
 FFNVCODEC_URL  = https://github.com/FFmpeg/nv-codec-headers/releases/download/n$(FFNVCODEC_VER)/nv-codec-headers-$(FFNVCODEC_VER).tar.gz
 FFNVCODEC_SHA1 = 89bee0d3025b813c513084ad8546b0261623fbb0
 
-FFMPEG         = ffmpeg-5.1.3
+FFMPEG         = ffmpeg-6.0
 FFMPEG_TB      = $(FFMPEG).tar.bz2
 FFMPEG_URL     = https://ffmpeg.org/releases/$(FFMPEG_TB)
-FFMPEG_SHA1    = 027cf7011307689fdb670317f5ff29eee7cfcfbf
+FFMPEG_SHA1    = 7ce79a1806f3f19b6b044f83d0cec39309dab0ee
 
 
 # ##############################################################################
diff -urN tvheadend/src/transcoding/transcode/hwaccels/vaapi.c tvheadend-ffmpeg6/src/transcoding/transcode/hwaccels/vaapi.c
--- tvheadend/src/transcoding/transcode/hwaccels/vaapi.c	2023-04-30 11:58:04.700160076 -0700
+++ tvheadend-ffmpeg6/src/transcoding/transcode/hwaccels/vaapi.c	2023-04-30 12:01:43.000000000 -0700
@@ -594,7 +594,9 @@
     }
 
     avctx->get_buffer2 = vaapi_get_buffer2;
+#if LIBAVCODEC_VERSION_MAJOR < 60
     avctx->thread_safe_callbacks = 0;
+#endif
 
     return 0;
 }
